src/components/App/PuzzleGenerate/index.tsx

```diff
@@ -25,87 +27,101 @@ const PuzzleGenerate = ({ puzzlesList }) => {
-            required={true}
-            placeholder="Longitude"
-            setInputVal={setLongitude}
-          />
-          <Input
-            type="text"
-            id="puzzleAddress"
-            required={true}
-            placeholder="Address"
-            setInputVal={setAddress}
-          />
-        </div>
-        <div>
-          {puzzlesList.length > 0 && (
+        <h3> Set New Puzzle Location </h3>
+        <div className={styles.selectionWrapper}>
+          {puzzlesList.length > 0 ? (
             <select
               className={styles.selections}
               value={puzzleId}
               onChange={e => setPuzzleId(e.currentTarget.value)}
             >
-              <option selected>Choose a puzzle</option>
-              {puzzlesList.map((puzzle, index) => (
+              <option>Choose a puzzle</option>
+              {puzzlesList.map((puzzle: PuzzleCustom, index: number) => (
                 <option value={puzzle.id} key={`puzzle${index}`}>
                   {puzzle.name}
                 </option>
               ))}
             </select>
+          ) : (
+            <>
+              <select className={styles.selections} value="No puzzle available">
+                \
+              </select>
+            </>
           )}
         </div>
+        <div className={styles.hintWrapper}>
+          <Input
+            type="text"
+            id="puzzleHint"
+            required={true}
+            placeholder="Hint"
+            setInputVal={setHint}
+          />
+        </div>
+        <div className={styles.addressWrapper}>
+          <Input
+            id="address-displayer"
+            type="text"
+            placeholder="Address"
+            value={address}
+            setInputVal={setAddress}
+            required={false}
+            // disabled={true}
+            onClick={() => setModalIsOpen(true)}
+          />
+          <Button
+            style="secondary"
+            content="Set Location"
+            size="sm"
+            type="button"
+            onClick={() => setModalIsOpen(true)}
+          />
+        </div>
         <div>
           <Button
             style="primary"
             type="submit"
-            size="sm"
             content={'Submit'}
             arrowDirection="right"
             onClick={() => handleSubmit()}
           />
         </div>
-        {puzzleInstanceData && (
-          <div className={styles.qrCode}>
-            {/** TODO: Create link to puzzle map page */}
-            {/** TODO: Make it copiable */}
-            <QRGenerator text={JSON.stringify(puzzleInstanceData)} />
-          </div>
-        )}
       </div>
+      <Modal
+        className={styles.modal}
+        isOpen={modalIsOpen}
+        onRequestClose={() => {
+          setModalIsOpen(false);
+        }}
+        contentLabel="Example Modal"
+        overlayClassName={styles.modalOverlay}
+        appElement={document.getElementById('__next') as HTMLElement}
```

Comment:
WHat does this do?

Reply:
This is the React Modal's thing, it require a context of what it's rendering on

Reply:
It's really weird, I got the same solution from github and StackOverflow, so it might be ligit